app/Http/Requests/CategoryRequest.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class CategoryRequest extends FormRequest
+{
+    public function authorize(): bool
+    {
+        return true;
+    }
+
+    public function rules(): array
+    {
+        return [
+            'name' => 'required|string|max:255',
+            'parent_id' => 'nullable|exists:categories,id',
```

Comment:
Please use Category::class instead of the categories table name to avoid hardcoding database table names in the code. This approach improves maintainability, as the table name can be defined or changed in the model itself if needed, without affecting the rest of the code.